### DISCUSSION TITLE
How to correctly place components in c4 diagram?

### DISCUSSION THREAD

**Discussion by boris-faniuk-n26** [mermaid-js/mermaid#4522] [Q&A]
Hello! 

I am playing with mermaid to build C4 diagrams in GitHub.
I cannot find any good way to place elements correctly.

I have tried putting 2 elements with a relation between them (see [live editor](https://mermaid.live/edit#pako:eNptjz8LwkAMxb_KERwUOjg4FVxscXPxEBSypG1qC727cpeKRfrdPSr-GdzyyO-9vDygdBVDCtkmc1b4LmjR6jEIm2Vgf2tLThTCYVT6pRBWbyAvlhUJFRTeTL6b10fuvuZfZM9SNio4w6o1vfNCVmZgtp36OHI0axk7_pewcHUdWM5bhDXCR19eOkZAAoa9obaKPz3QqnhUGjaxdhrHimsaOkFAO0WUBnF6tCWk4gdOYJgL5C1dPRlIa-oCT08bLWOY)).
Both elements are located so close to each that relation text is not visible.
Is there any way to improve this?

I have also tried adding empty element with white background (see [live editor](https://mermaid.live/edit#pako:eNptkE1LxEAMhv_KEPZQocIieyp4sdWbF4ugkEvaSbeF-SgzqVqW_e-OU9RFhBySd948SeYEvdcMFdSH2jvhD0GHrl2jsC0ih7ep51IhPK6q3SqEK3S1t7N37KRgO8taIlDUW0TKjg3RdIUmoY7iN6W5y89PbH7xl5YHln5U0VtWU5oRhJxkQ257nlPKqbmV1fB_hJ0fhsjycotwfdgj_CivX8rN_gJzb9imEzbUdofadcfaGx-S-X2chDOg80Fz-KMnDJRgOViadPrAEzqV9pcxQRGqlGoeaDGCgO6crLSIb1fXQyVh4RKWvEQz0TGQhWogE_n8CSYXilo)).
However, then we depend on the background colour that is different across GitHub themes.
Anyway, empty element which requires it's own UpdateElementStyle block seems as workaround for me.
I hope there's a better native solution for this.

Thanks for any help.

**Comment by ghost** [comment]
The layout support for C4 diagrams is really poor from what I can tell. The most I've seen is that you can limit the number of entities in a row before it wraps, but the diagrams it produces are not easy to interpret. I'd love to see support for the layout hints that the core PlantUML C4 libraries use as grouping and layout is important to telling an understandable story—as it stands the diagrams it produces are no better than trying to achieve the same thing with text.
